src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -447,36 +459,38 @@ void productionGroup(const Opm::Schedule&     sched,
         return;
     }
 
-    if (cgroup && cgroup->name() == "FIELD")
-        throw std::logic_error("Got cgroup == FIELD - uncertain logic");
-    // group is available for higher level control, but is currently constrained by own limits
-    iGrp[nwgmax + IGroup::ProdHighLevCtrl] = -1;
-    if ((deck_cmode != Opm::Group::ProductionCMode::FLD) && !group.productionGroupControlAvailable()) {
-        //group is not free to respond to higher level control)
+    if (group.name() == "FIELD" ) {
         iGrp[nwgmax + IGroup::ProdHighLevCtrl] = 0;
-    } else if (cgroup && ((active_cmode == Opm::Group::ProductionCMode::FLD) || (active_cmode == Opm::Group::ProductionCMode::NONE))) {
-        //a higher level group control is active constraint
-        if ((deck_cmode != Opm::Group::ProductionCMode::FLD) && (deck_cmode != Opm::Group::ProductionCMode::NONE)) {
-            iGrp[nwgmax + IGroup::ProdHighLevCtrl] = cgroup->insert_index();
-        } else if ((deck_cmode == Opm::Group::ProductionCMode::FLD) && (prod_guide_rate_def != Opm::Group::GuideRateProdTarget::NO_GUIDE_RATE)) {
-            iGrp[nwgmax + IGroup::ProdHighLevCtrl] = cgroup->insert_index();
-        } else if ((deck_cmode == Opm::Group::ProductionCMode::NONE) && group.productionGroupControlAvailable() &&
-                   (prod_guide_rate_def != Opm::Group::GuideRateProdTarget::NO_GUIDE_RATE)) {
-            iGrp[nwgmax + IGroup::ProdHighLevCtrl] = cgroup->insert_index();
+    } else {
+        // group is available for higher level control, but is currently constrained by own limits
+        iGrp[nwgmax + IGroup::ProdHighLevCtrl] = -1;
+        if ((deck_cmode != Opm::Group::ProductionCMode::FLD) && !group.productionGroupControlAvailable()) {
+            //group is not free to respond to higher level control)
+            iGrp[nwgmax + IGroup::ProdHighLevCtrl] = 0;
+        } else if (cgroup && ((active_cmode == Opm::Group::ProductionCMode::FLD) || (active_cmode == Opm::Group::ProductionCMode::NONE))) {
+            //a higher level group control is active constraint
+            if ((deck_cmode != Opm::Group::ProductionCMode::FLD) && (deck_cmode != Opm::Group::ProductionCMode::NONE)) {
+                iGrp[nwgmax + IGroup::ProdHighLevCtrl] = cgroup->insert_index();
+            } else if ((deck_cmode == Opm::Group::ProductionCMode::FLD) && (prod_guide_rate_def != Opm::Group::GuideRateProdTarget::NO_GUIDE_RATE)) {
```

Comment:
This is heacy reading - is it possible to summarize in a few sentences of english?

Reply:
The main changes are to handle the FIELD group level control correctly. That consists of two main changes:

1. The controlling group (production / injection) can be the FIELD level as well as a higher group level (controlGroup and injectionControlGroup functions)

2. A group being controlled by a group at a higher level can also be controlled from the "FIELD" level.